trieste/acquisition/sampler.py

```diff
@@ -276,3 +279,218 @@ def sample(self, at: TensorType, *, jitter: float = DEFAULTS.JITTER) -> TensorTy
         new_order = tf.concat([leading_indices, absolute_trailing_indices], axis=0)
 
         return mean[..., None, :, :] + tf.transpose(variance_contribution, new_order)
+
+
+TrajectoryFunction = Callable[[TensorType], TensorType]
+"""
+Type alias for trajectory functions.
+
+An :const:`TrajectoryFunction` evaluates a particular sample at a set of `N` query
+points (each of dimension `D`) i.e. takes input of shape `[N, D]` and returns
+shape `[N, 1]`.
+
+A key property of these trajectory functions is that the same sample draw is evaluated
+for all queries. This property is known as consistency.
+"""
+
+
+class ContinuousSampler(ABC):
+    r"""
+    An :class:`ContinuousSampler` samples a specific quantity according
+    to an underlying :class:`ProbabilisticModel`.
+    Unlike our :class:`DiscreteSampler`, :class:`ContinuousSampler` returns a
+    queryable function (a trajectory) that provides the sample's value at a query location.
+    """
+
+    def __init__(self, dataset: Dataset, model: ProbabilisticModel):
```

Comment:
Is it a good idea that we can also have a `sample_size` arg here, so that sample method(i.e., `get_trajectory`) can return a `list[TrajectoryFunction]`  for instance?

Reply:
I think its fine just to call this multiple times if you want multiple samples. Otherwise we have to deal with varied sized outputs (i.e. sometimes just a function, sometimes a list of functions).